### DISCUSSION TITLE
Unit test in CodeSys

### DISCUSSION THREAD

**Discussion by HenrikJuulsgaard** [tcunit/TcUnit#191] [Q&A] [answered]
Hi!

Anyone tried to use TcUnit in CodeSys IDE?

**Comment by sagatowski** [comment, marked as answer]
Check https://forge.codesys.com/lib/counit/home/Home/

**Reply by HAHermsen** [reply]
Thanks @sagatowski for forwarding and thank @HenrikJuulsgaard for asking.

TcUnit for TwinCAT is dubbed coUnit for CODESYS. 
We aim to be as close as necessary to the original TcUnit codebase, due to obvious reasons.

The CODESYS fork shares >85% of the original TcUnit codebase but due to technical differences between TwinCAT and CODESYS we have some notable differences with TcUnit which are clearly highlighted in the coUnit documentation.

With kind regards,


Haico Hermsen


https://forge.codesys.com/lib/counit/home/Home/
https://www.co5e.org

**Reply by HenrikJuulsgaard** [reply]
Thank's Haico, just what we needed!
I'll start hacking, looks like the documentation will give us a smooth liftoff.
